matRad_singleBeamView.m

```diff
@@ -0,0 +1,172 @@
+function sb_resultGUI = matRad_singleBeamView(pln, cst, weights, viewBeamNum, dij, ct, stf)
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%  calculate single beam dose distributions
+%   this can be done either by providing a dij matrix or by calculating the
+%   single beam dose directly from the corresponding spotweights
+%  
+%
+% call
+%   resultGUI = matRad_singleBeamView(pln, cst, resultGUI.w, viewBeamNum, dij)
+%   or
+%   resultGUI = matRad_singleBeamView(pln,cst,resultGUI.w,viewBeamNum,[],ct,stf)
+%
+% input
+%   pln:         matRad pln struct
+%   cst:         matRad cst struct
+%   weights:     weights for beam spots (matRad resultGUI.w)
+%   viewBeamNum: beam to be visualised, either integer for beamNum or 'all'
+%                to calculate total dose
+%
+%   dij:         matRad dij struct (optional, single beam dose calculation
+%                will use this, otherwise pass empty matrix)
+%               
+%                if you provide these parameters dose will be calculated
+%                directly from spotweights:
+%   ct:          matRad ct struct (optional, only needed if no dij provided)
+%   stf:         matRad stf struct (optional, only needed if no dij provided)
+%
+% output
+%  resultGUI:    matRad resultGUI struct containing single beam dose of
+%                beam viewBeamNum
+% References
+%   -
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+if ~isempty(dij)
+    % calculate single beam dose from dij
+    if strcmp(viewBeamNum, 'all')
+        % recalculate standard resultGUI for all beams
+        fprintf('Calculating total dose \n');
+        sb_resultGUI = matRad_calcCubes(weights,dij,cst,1);
+
+    elseif ~isnumeric(viewBeamNum)
+        error('wrong beam number format in matRad_singleBeamView \n');
+
+    else
+        % calculate single beam dose
+        
+        % adjust cst for single beams
+        sb_cst = cst;
+        for i=1:size(sb_cst,1)
```

Comment:
I thought since matRad_calcCubes expects a cst-file it might make more sense to pass a consistent cst eventhough the actual dose information is not used.